sklearnex/decomposition/pca.py

```diff
@@ -15,4 +15,314 @@
+            self.tol = tol
+            self.iterated_power = iterated_power
+            self.n_oversamples = n_oversamples
+            self.power_iteration_normalizer = power_iteration_normalizer
+            self.random_state = random_state
+
+        def fit(self, X, y=None):
+            self._fit(X)
+            return self
+
+        def _fit(self, X):
+            if sklearn_check_version("1.2"):
+                self._validate_params()
+            elif sklearn_check_version("1.1"):
+                check_scalar(
+                    self.n_oversamples,
+                    "n_oversamples",
+                    min_val=1,
+                    target_type=numbers.Integral,
+                )
+
+            X = self._validate_data(
+                X,
+                dtype=[np.float64, np.float32],
+                ensure_2d=True,
+                copy=self.copy,
+            )
+
+            U, S, Vt = dispatch(
+                self,
+                "fit",
+                {
+                    "onedal": self.__class__._onedal_fit,
+                    "sklearn": sklearn_PCA._fit,
+                },
+                X,
+            )
+            return U, S, Vt
+
+        @run_with_n_jobs
+        def _onedal_fit(self, X, queue=None):
+            if issparse(X):
+                raise TypeError(
+                    "PCA does not support sparse input. See "
+                    "TruncatedSVD for a possible alternative."
+                )
+
+            self._validate_n_components(self.n_components, X.shape[0], X.shape[1])
+
+            onedal_params = {
+                "n_components": self.n_components,
+                "is_deterministic": True,
+                "method": "cov",
+                "whiten": self.whiten,
+            }
+            self._onedal_estimator = onedal_PCA(**onedal_params)
+            self._onedal_estimator.fit(X, queue=queue)
+            self._save_attributes()
+
+            U = None
+            S = self.singular_values_
+            Vt = self.components_
+
+            return U, S, Vt
+
+        @wrap_output_data
+        def transform(self, X, y=None):
+            X = self._validate_data(
+                X,
+                dtype=[np.float64, np.float32],
+                reset=False,
+            )
+
+            return dispatch(
+                self,
+                "transform",
+                {
+                    "onedal": self.__class__._onedal_transform,
+                    "sklearn": sklearn_PCA.transform,
+                },
+                X,
+            )
+
+        @run_with_n_jobs
+        def _onedal_transform(self, X, queue=None):
+            check_is_fitted(self)
+
+            self._validate_n_features_in_after_fitting(X)
+            if sklearn_check_version("1.0"):
+                self._check_feature_names(X, reset=False)
+
+            return self._onedal_estimator.predict(X, queue=queue)
+
+        def fit_transform(self, X, y=None):

```

Comment:
https://github.com/intel/scikit-learn-intelex/blob/46ba021e483b24098e4ef0776266869877f9d8cc/onedal/_device_offload.py#L62

```suggestion
        @support_usm_ndarray()
        def fit_transform(self, X, y=None):

```

Reply:
Please wrap out this public method with @support_usm_ndarray decorator

Reply:
Sorry, for `support_usm_ndarray ` it is only for onedal4py estimators. Using `wrap_output_data` will be ok.